contracts/src/op/OPSuccinctDisputeGame.sol

```diff
@@ -0,0 +1,362 @@
+        if (l2BlockNumber() <= rootBlockNumber)
+            revert UnexpectedRootClaim(rootClaim());
+
+        // Set the root claim
+        claimData = ClaimData({
+            parentIndex: type(uint32).max,
+            counteredBy: address(0),
+            claimant: gameCreator(),
+            bond: uint128(msg.value),
+            claim: rootClaim(),
+            clock: LibClock.wrap(
+                Duration.wrap(0),
+                Timestamp.wrap(uint64(block.timestamp))
+            )
+        });
+
+        // Set the game as initialized.
+        initialized = true;
+
+        // Deposit the bond.
+        credit[gameCreator()] = msg.value;
+
+        // Set the game's starting timestamp
+        createdAt = Timestamp.wrap(uint64(block.timestamp));
+    }
+
+    ////////////////////////////////////////////////////////////
+    //                    IDisputeGame impl                   //
+    ////////////////////////////////////////////////////////////
+
+    /// @notice Getter for the game type.
+    /// @dev The reference impl should be entirely different depending on the type (fault, validity)
+    ///      i.e. The game type should indicate the security model.
+    /// @return gameType_ The type of proof system being used.
+    function gameType() public pure returns (GameType) {
+        // TODO(fakedev9999): Once a new version of the Optimism contracts containing the PR below is released,
+        // update this to return the correct game type: GameTypes.OP_SUCCINCT
+        // https://github.com/ethereum-optimism/optimism/pull/13780
+        return GameType.wrap(6);
+    }
+
+    /// @notice Getter for the creator of the dispute game.
+    /// @dev `clones-with-immutable-args` argument #1
+    /// @return The creator of the dispute game.
+    function gameCreator() public pure returns (address) {
+        return _getArgAddress(0x00);
+    }
+
+    /// @notice Getter for the root claim.
+    /// @dev `clones-with-immutable-args` argument #2
+    /// @return The root claim of the DisputeGame.
+    function rootClaim() public pure returns (Claim) {
+        return Claim.wrap(_getArgBytes32(0x14));
+    }
+
+    /// @notice Getter for the parent hash of the L1 block when the dispute game was created.
+    /// @dev `clones-with-immutable-args` argument #3
+    /// @return The parent hash of the L1 block when the dispute game was created.
+    function l1Head() public pure returns (Hash) {
+        return Hash.wrap(_getArgBytes32(0x34));
+    }
+
+    /// @notice Getter for the extra data.
+    /// @dev `clones-with-immutable-args` argument #4
+    /// @return extraData_ Any extra data supplied to the dispute game contract by the creator.
+    function extraData() public pure returns (bytes memory extraData_) {
+        // The extra data starts at the second word within the cwia calldata and
+        // is 32 bytes long.
+        extraData_ = _getArgBytes(0x54, 0x20);
+    }
+
+    function challengeWithProof(
```

Comment:
Add description for this.

Reply:
tl;dr Initialize game -> call `challengeWithProof` -> `resolve` within the period before the proof is finalized.